src/iuse.cpp

```diff
@@ -5567,14 +5567,12 @@ int iuse::toolmod_attach( player *p, item *it, bool, const tripoint& ) {
         return 0;
     }
 
-    if( !loc->ammo_remaining() || g->unload( *loc ) ) {
-        //~ $1 - toolmod, $2 - base item
-        p->add_msg_if_player( m_good, _( "You attach the %1$s to the %2$s" ),
-                              it->tname().c_str(), loc->tname().c_str() );
-
-        loc->contents.push_back( p->i_rem( it ) );
+    if( loc->ammo_remaining() ) {
+        g->unload( *loc );
```

Comment:
The result of `unload` is now ignored. It is actually important as it may be `false`, which indicates the user has canceled the unloading (e.g. because they don't have a suitable container for liquid ammo).

Also: why did you remove the comment (starting with `~` means it's for the translators)?

Also: why are you now calling `add_msg` instead of `add_msg_if_player`?